lib/internal/http2/util.js

```diff
@@ -415,16 +417,18 @@ function mapToHeaders(map,
       if (isArray) {
         for (var k = 0; k < value.length; k++) {
           val = String(value[k]);
-          ret.push([key, val]);
+          ret += `${key}\0${val}\0`;
         }
+        count += value.length;
       } else {
         val = String(value);
-        ret.push([key, val]);
```

Comment:
@jasnell I do like the second approach, for the reason you mentioned. :) Do you think we might ever want to add more options than `NGHTTP2_NV_FLAG_NO_INDEX`? I don’t the `NO_COPY` flags would be set by userland, so we’re good on that, right?